dev_tools/ownership_utils.py

```diff
@@ -16,8 +16,11 @@ def __init__(self, pattern: str, owners: Tuple[str, ...], line_number: int) -> N
 
 
 class GithubOwnerShip:
-    def __init__(self, repo_dir: Path) -> None:
-        self._ownerships = parse_ownership(repo_dir / ".github" / "CODEOWNERS")
+    def __init__(self, repo_dir: Path, codeowners_file: Path = Path(".github") / "CODEOWNERS") -> None:
+        codeowners_resolved_path = (
+            codeowners_file if codeowners_file.is_absolute() else (repo_dir / codeowners_file).resolve()
```

Comment:
I think the relative_to approach makes more sense since it will raise an exception if the codeowners file is outside of the repo

Reply:
But there's a problem with your previous suggestion, after fixing order of arguments. If `codeowners_file` is relative, it will be relative to the current working dir. But I want it to be relative to the repo_dir aka external repo (chicken and egg problem).
```
        self._ownerships = parse_ownership(repo_dir / codeowners_file.relative_to(repo_dir.resolve()))
```

Reply:
That's only for your case of testing with `pre-commit try-repo` (we can find a different solution for that). But to me it does not make sense to allow the codeowners file outside of the repo.

Reply:
> it does not make sense to allow the codeowners file outside of the repo.
agreed, I adapted to that

> testing with pre-commit try-repo
Actually `poetry run check-ownership` works better because I can pass args. I proposed an update to README with the command.